docs/cloud/tutorials/gateway-end-user.mdx

```diff
@@ -0,0 +1,184 @@
+---
+sidebar_position: 60
+title: "Authenticating End Users with Gateway"
+description: "A tutorial on authenticating end users using the WebSocket Gateway"
+---
+
+This tutorial will walk you through setting up multi-user or multi-tenant access to a topic in Infinyon Cloud.
+
+## Concepts
+
+* **WebSocket Gateway**: The WebSocket Gateway allows you to produce and consume data using a WebSocket connection.
+* **Access Key**: Used to authenticate the WebSocket connection. This key will only allow access to a cluster and topic specified by the Access Key. This key can be shared with multiple end users.
+* **End-User Token**: A token that can be used to authenticate a end-user in addition to the Access Key. This token can be used to filter Fluvio records based on the user's identity.
+* **Fluvio Record Key**: A record in Fluvio is a key-value pair. The key can be set when producing, and can contain metadata such as end user identity.
+* **Key Filter**: A string that is used to filter records being consumed from the WebSocket Gateway. This filter can be used to restrict access to records based on the end user's identity. This value is checked against the Fluvio Record Key for an exact match.
+* **Key Filter Server**: A HTTP server that exposes a HTTP POST endpoint which returns a Key Filter based on End-User Token.
+* **Key Filter Remote URL**: The HTTPS URL of the Key Filter Server.
+
+## Prerequisites
+
+This tutorial assumes that the Fluvio CLI is installed, and logged-in to InfinyOn Cloud. Follow the [Quick Start] to get set up.
+
+## Create a topic
+
+First, create a topic to interact with.
+
+```shell copy="fl"
+$ fluvio topic create multi-user-demo
+topic "multi-user-demo" created
+```
+
+## Produce records
+
+Produce end user keyed records to the topic using the Fluvio CLI.
+
+```shell copy="cmd"
+$ echo "Message for User1" | fluvio produce multi-user-demo --key "user1"
+$ echo "Message for User2" | fluvio produce multi-user-demo --key "user2"
+```
+
+Confirm that the records were produced successfully with the correct key.
+
+```shell copy="fl"
+$ fluvio consume -Bdk multi-user-demo
+Consuming records from 'multi-user-demo' starting from the beginning of log
+[user1] Message for User1
+[user2] Message for User2
+```
+
+## Set up a Key Filter Server (Rust Example)
+
+Create a simple Key Filter Server that returns a Key Filter based on the End-User Token.
+
+### Setup a new Rust project
+
+Create a HTTP application server using [Actix].
+
+```shell copy="fl"
+$ cargo new key_filter_server
+```
+
+Add the `actix-web` dependency to the `Cargo.toml` file.
+
+```toml
```

Comment:
move code examples to embedded

Reply:
how? is there an example?

Reply:
https://raw.githubusercontent.com/infinyon/fluvio-docs/refs/heads/main/docs/fluvio/apis/rust/example.mdx

Reply:
done